public/app/plugins/datasource/prometheus/language_utils.ts

```diff
@@ -1,10 +1,18 @@
+import { PromMetricsMetadata } from './types';
+
 export const RATE_RANGES = ['1m', '5m', '10m', '30m', '1h'];
+export const MAX_LABEL_LENGTH = 1000;
 
 export const processHistogramLabels = (labels: string[]) => {
   const result = [];
   const regexp = new RegExp('_bucket($|:)');
   for (let index = 0; index < labels.length; index++) {
-    const label = labels[index];
+    let label = labels[index];
+
+    if (label.length > MAX_LABEL_LENGTH) {
+      label = label.substr(0, MAX_LABEL_LENGTH);
+    }
+
     const isHistogramValue = regexp.test(label);
```

Comment:
Not sure how important this is in this case but the regex basically checks whether it ends with `_bucket` or has `_bucket:` in either case this will probably change the outcome in case of truncated label.